drgn/helpers/linux/wait.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+Waitqueue
+--------------
+
+The ``drgn.helpers.linux.waitq`` module provides helpers for working with the
+wait-queue mechanism, implemented in :linux:`include/linux/wait.h`
+(``struct wait_queue_head`` and ``struct wait_queue_entry``).
+"""
+
+from typing import Iterator
+
+from drgn import Object, cast
+from drgn.helpers.linux.list import list_empty, list_for_each_entry
+
+__all__ = (
+    "waitqueue_active",
+    "waitqueue_for_each_entry",
+    "waitqueue_for_each_task",
+)
+
+
+def _get_wait_queue_head(waitq: Object) -> Object:
```

Comment:
In the kernel APIs, the wait queue always seems to be named `wq`, so can we do that everywhere in this file, too?

```suggestion
def _get_wait_queue_head(wq: Object) -> Object:
```

Reply:
Sure thing. I have made the changes.